helix-term/src/ui/menu.rs

```diff
@@ -100,14 +105,16 @@ impl<T: Item> Menu<T> {
                 .enumerate()
                 .filter_map(|(index, option)| {
                     let text = option.filter_text(&self.editor_data);
+                    let group = option.group(&self.editor_data);
                     // TODO: using fuzzy_indices could give us the char idx for match highlighting
                     self.matcher
                         .fuzzy_match(&text, pattern)
-                        .map(|score| (index, score))
+                        .map(|score| (index, group, score))
                 }),
         );
         // Order of equal elements needs to be preserved as LSP preselected items come in order of high to low priority
-        self.matches.sort_by_key(|(_, score)| -score);
+        self.matches
+            .sort_by_key(|(_, group, score)| (*group, -*score));
```

Comment:
Ah actually if we keep this code as is then it actually shows all items of LSP A over LSP B that does need a cobfig option of sorts. I think it would be the other way around. This should not be the default iMO